packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,352 @@
+      next: undefined,
+      prev: undefined,
+      parent: divNode,
+    };
+    divNode.children?.push(imgNode);
+    child = divNode;
+  } else if (isOcticon) {
+    const octiconIcon = getOcticonIcon(iconName!);
+    let svgStr = octiconIcon.toSVG({
+      class: octiconClass ? `${iconClass} ${octiconClass} ${className}` : `${iconClass} ${className}`,
+    });
+
+    if (octiconColor) {
+      svgStr = svgStr.replace('<path', `<path fill="${octiconColor}"`);
+    }
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: [
+        {
+          type: 'text',
+          data: svgStr,
+        },
+      ],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  } else {
+    child = {
+      type: 'tag',
+      name: 'span',
+      attribs: {
+        class: className ? `${icon} ${className}` : icon,
+        'aria-hidden': 'true',
+        style: `${defaultSize}margin-right:5px;${size.fontSize ? `font-size:${size.fontSize};` : ''}`,
+      },
+      children: iconName ? [
+        {
+          type: 'text',
+          data: iconName,
+        },
+      ] : [],
+      next: undefined,
+      prev: undefined,
+      parent,
+    };
+  }
+  return child;
+}
+
+function getSize(widthPar: string, heightPar: string) {
+  const width = widthPar || heightPar || '25px';
+  const height = heightPar || widthPar || '25px';
+  const sizeUnitRegexp = /(\d+)(px|em|rem|%|vw|vh|vmin|vmax|ex|ch)/;
+  const widthMatch = width.match(sizeUnitRegexp);
+  const heightMatch = height.match(sizeUnitRegexp);
+
+  if (!widthMatch || !heightMatch) {
+    throw new Error(`Invalid width or height format. 
+                    Expected format is number followed by unit
+                    (px | em | rem |%| vw | vh | vmin | vmax | ex | ch).`);
+  }
+
+  // Ensure both width and height are using the same unit
+  if (widthMatch[2] !== heightMatch[2]) {
+    throw new Error('The units of width and height should be the same.');
+  }
+
+  let fontSize = Math.min(parseFloat(widthMatch[1]), parseFloat(heightMatch[1]));
+
+  // Assuming the single character font will take up approximately 70% of the container's smaller dimension
+  fontSize *= 0.7;
+
+  return {
+    fontSize: `${fontSize}${widthMatch[2]}`, // fontSize in the same unit as width and height
+    width,
+    height,
+  };
+}
+
+function createDivChild(parent: NodeOrText, children: NodeOrText[]): NodeOrText {
+  return {
+    type: 'tag',
+    name: 'div',
+    attribs: {},
+    children,
+    next: undefined,
+    prev: undefined,
+    parent,
+  };
+}
+
+function updateNodeStyle(node: NodeOrText) {
+  const styleObject = getStyleObject(node);
+  styleObject['list-style-type'] = 'none';
+  styleObject['padding-left'] = '0';
+  if (!node.attribs) node.attribs = {};
+
+  node.attribs.style = createStyleString(styleObject);
```

Comment:
Its good you are attempting to eliminate duplicate properties (if that's the intention), but I don't think it is worth the extra maintainence and readability cost.

Reply:
alternative: https://cheerio.js.org/docs/api/classes/Cheerio#css

Reply:
> +  node.attribs.style = node.attribs.style + 'list-style-type: none; padding-left: 0;';

node.attribs.style = node.attribs.style + 'list-style-type: none; padding-left: 0;';
If the node already has a custom style, this action will override it.
This code aims to maintain the current style while removing the bullet points.

Reply:
Is the old style not concatenated with `node.attribs.style + `?

To be more explicit:
```
node.attribs.style = (node.attribs.style || '') + '; list-style-type: none; padding-left: 0;';
```

Reply:
You can use cheerio's method instead as well to remove the low level handling if you prefer to avoid duplicate properties: https://cheerio.js.org/docs/api/classes/Cheerio#css

Reply:
I see what you mean by duplication, thanks for the suggested improvement.